test/pylib/scylla_cluster.py

```diff
@@ -113,6 +113,61 @@ def make_scylla_conf(workdir: pathlib.Path, host_addr: str, seed_addrs: List[str
     '--abort-on-ebadf', '1'
 ]
 
+# [--smp, 1], [--smp, 2] -> [--smp, 2]
+# [--smp, 1], [--smp] -> [--smp]
+# [--smp, 1], [--smp, __missing__] -> [--smp]
+# [--smp, 1], [--smp, __remove__] -> []
+# [--smp=1], [--smp=2] -> [--smp, 2]
+# [--smp=1], [--smp=__remove__] -> []
+# [--overprovisioned, --smp=1, --abort-on-ebadf], [--smp=2] -> [--overprovisioned, --smp=2, --abort-on-ebadf]
+# [], [--experimental-features, raft, --experimental-features, broadcast-tables] ->
+# [--experimental-features, raft, --experimental-features, broadcast-tables]
+def merge_cmdline_options(base: List[str], override: List[str]) -> List[str]:
+    if len(override) == 0:
+        return base
+
+    def to_dict(args: List[str]) -> Dict[str, List[str]]:
+        result: Dict[str, List[str]] = {}
```

Comment:
Why is there a list of values for a key instead of a single value? What is the use case for this? In all examples you provided there at most one value for each key, so all lists you end up are of the form `[v]` where `v: str` or `[None]`. Wouldn't `Optional[str]` be simpler?

Reply:
There is a specific example for this `# [--experimental-features, raft, --experimental-features, broadcast-tables]`.
In general, parameter can have many values.